src/hamiltonian.jl

```diff
@@ -70,15 +70,18 @@ phasepoint(
 
 # If position variable and momentum variable are in different containers,
 # move the momentum variable to that of the position variable.
-# This is needed for AHMC to work with CuArrays (without depending on it).
-phasepoint(
+# This is needed for AHMC to work with CuArrays and other Arrays (without depending on it).
+function phasepoint(
```

Comment:
Can you implement your change while keeping the current interface of optionally provide `ℓπ` and `ℓκ`?
You can create a function for your version of initializing `r`, e.g. `safe_rsimilar(θ, _r) = (r = similar(θ); copyto!(r, _r); r)`, and update `r=safe_rsimilar((θ, _r))`